src/aligners/nw.rs

```diff
@@ -8,10 +8,66 @@ use crate::prelude::{Pos, Sequence, I};
 use std::cmp::min;
 use std::iter::zip;
 
+pub type PATH = Vec<(usize, usize)>;
 pub struct NW<CostModel> {
     pub cm: CostModel,
 }
 
+#[derive(PartialEq)]
+enum CIGAR_option {
+    Match,
+    Mismatch,
+    Insertion,
+    Deletion,
+}
+
+pub struct CIGAR_obj {
+    command: CIGAR_option,
+    n: usize,
+}
+
+fn new_CIGAR(command: CIGAR_option) -> CIGAR_obj {
+    CIGAR_obj {
+        command: command,
+        n: 1,
+    }
+}
+
+pub type CIGAR = Vec<CIGAR_obj>;
+
+fn cigar_add(cigar: &mut CIGAR, command: CIGAR_option) {
+    if let Some(s) = cigar.last_mut() {
+        if s.command == command {
+            s.n += 1;
+            return;
+        }
+    }
+    cigar.push(new_CIGAR(command));
+}
+
+fn get_CIGAR_char(command: &CIGAR_option) -> char {
+    match command {
+        Match => 'M',
```

Comment:
This is super confusing, but:
- the compilr currently thinks this is a variable name, as in:
```
match command {
   x => { do something with x }
}
```
Instead, you want `GIGAR_option::Match` here.
Or you can do `use CIGAR_optIon::*` somewhere, but I think `Insert`/`Match`/.. are a bit too general to use everywhere